app/controllers/hiring_staff/vacancies_controller.rb

```diff
@@ -80,13 +86,20 @@ def step_valid?(step_form)
   end
 
   def redirect_to_incomplete_step
-    return redirect_to organisation_job_job_specification_path(@vacancy.id) unless step_valid?(JobSpecificationForm)
-    return redirect_to organisation_job_pay_package_path(@vacancy.id) unless step_valid?(PayPackageForm)
-    return redirect_to organisation_job_important_dates_path(@vacancy.id) unless step_valid?(ImportantDatesForm)
-    return redirect_to organisation_job_supporting_documents_path(@vacancy.id) unless
-      step_valid?(SupportingDocumentsForm)
-    return redirect_to organisation_job_application_details_path(@vacancy.id) unless step_valid?(ApplicationDetailsForm)
-    return redirect_to organisation_job_job_summary_path(@vacancy.id) unless step_valid?(JobSummaryForm)
+    computer_readable_step_names.each do |step|
```

Comment:
This is very nice, but does it make the code better? I'd argue it obfuscates the intention of the code and makes it less readable. If the pros > cons let's do it, but I'd like to hear what others think about it 👍

Reply:
I probably got carried away here. Not sure if it makes code better - prevents a very small need for upkeep if we add more steps to hiring staff journey.

Reply:
Reverted